File: PageObjects/EbayHome.cs
using NUnit.Framework;
using NUnit;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium.Interactions;

namespace selenium_assignment.PageObjects
{
    class EbayHome
    {
        private IWebDriver _driver;

        public EbayHome(IWebDriver driver)
        {
            _driver = driver;
        }


        public By searchField=By.XPath("//input[@type='text']");
        public By btn_AllCategories=By.XPath("//div[@id='gh-cat-box']/child::select");
        public By category_ConsumerElectronics=By.XPath("//div[@id='gh-cat-box']/child::select/child::option[text()='Consumer Electronics']");
        public By btn_Search=By.XPath("//input[@class='btn btn-prim gh-spr']");
    

        //Fuctions to test if elements are being displayed 
        public bool exists_searchField()
        {
            return _driver.FindElement(searchField).Displayed;
        }
        public bool exists_btn_AllCategories()
        {
            return _driver.FindElement(btn_AllCategories).Displayed;
        }
        public bool exists_category_ConsumerElectronics()
        {
            return _driver.FindElement(category_ConsumerElectronics).Displayed;
        }
        public bool exists_btn_Search()
        {
            return _driver.FindElement(btn_Search).Displayed;
        }
       
        
        public void searchInput(string input)
        {
            _driver.FindElement(searchField).SendKeys(input);
        }
       
       

    }
}

File: python/Pages/EbayHome.py
class EbayHome:

    def __init__(self, driver):
        self.driver=driver

    searchField = "//input[@type='text']"
    btn_AllCategories = "//div[@id='gh-cat-box']/child::select"
    category_ConsumerElectronics = "//div[@id='gh-cat-box']/child::select/child::option[text()='Consumer Electronics']"
    btn_Search = "//input[@class='btn btn-prim gh-spr']"

    def searchInput(self, input):
        self.driver.find_element_by_xpath(self.searchField).send_keys(input)

    def displaysearchField(self):
        return self.driver.find_element_by_xpath(self.searchField).is_displayed()

    def displaybtn_AllCategories(self):
        return self.driver.find_element_by_xpath(self.btn_AllCategories).is_displayed()

    def displaycategory_ConsumerElectronics(self):
        return self.driver.find_element_by_xpath(self.category_ConsumerElectronics).is_displayed()

    def displaybtn_Search(self):
        return self.driver.find_element_by_xpath(self.btn_Search).is_displayed()

    def clickbtn_AllCategories(self):
        self.driver.find_element_by_xpath(self.btn_AllCategories).click()

    def clickcategory_ConsumerElectronics(self):
        self.driver.find_element_by_xpath(self.category_ConsumerElectronics).click()

    def clickbtn_Search(self):
        self.driver.find_element_by_xpath(self.btn_Search).click()
